core/domain/exp_jobs_one_off.py

```diff
@@ -506,3 +506,45 @@ def map(exp_summary_model):
     def reduce(key, values):
         """Implements the reduce function for this job."""
         yield (key, len(values))
+
+
+class RegenerateMissingExpCommitLogModels(jobs.BaseMapReduceOneOffJobManager):
+    """Job that regenerates missing commit log models for an exploration."""
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        rights_model = exp_models.ExplorationRightsModel.get_by_id(item.id)
+
+        for version in python_utils.RANGE(1, item.version + 1):
+            commit_log_model = (
+                exp_models.ExplorationCommitLogEntryModel.get_by_id(
+                    'exploration-%s-%s' % (item.id, version)))
+            if commit_log_model is None:
+                metadata_model = (
+                    exp_models.ExplorationSnapshotMetadataModel.get_by_id(
+                        '%s-%s' % (item.id, version)))
+                commit_log_model = (
+                    exp_models.ExplorationCommitLogEntryModel.create(
+                        item.id, version, metadata_model.committer_id,
+                        metadata_model.commit_type,
+                        metadata_model.commit_message,
+                        metadata_model.commit_cmds,
+                        rights_model.status, rights_model.community_owned))
+                commit_log_model.exploration_id = item.id
+                commit_log_model.put()
+                yield ('Regenerated Exploration Commit Log Model', item.id)
+
+            if commit_log_model.deleted:
```

Comment:
We should not be reverting deleted status -- if this is an issue that occurs in practice, it is the sort of thing that needs to be looked into further to figure out the cause, so that we can determine the appropriate fix.

Reply:
This does not occur (If a commit log model is marked as deleted and we do a commit after that, [error is raised automatically](https://github.com/oppia/oppia/blob/7d8e659264582d7ce74bc6c139e597b82bca0e04/core/storage/base_model/gae_models.py#L597)) - I have removed it.